tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx

```diff
@@ -75,31 +132,39 @@ public:
    RNTupleInspector &operator=(RNTupleInspector &&other) = delete;
    ~RNTupleInspector() = default;
 
-   /// Creates a new inspector for a given RNTuple.
-   static RResult<std::unique_ptr<RNTupleInspector>>
-   Create(std::unique_ptr<ROOT::Experimental::Detail::RPageSource> pageSource);
-   static RResult<std::unique_ptr<RNTupleInspector>> Create(RNTuple *sourceNTuple);
-   static RResult<std::unique_ptr<RNTupleInspector>> Create(std::string_view ntupleName, std::string_view storage);
+   /// Create a new inspector for a given RNTuple.
+   static std::unique_ptr<RNTupleInspector> Create(std::unique_ptr<Detail::RPageSource> pageSource);
+   static std::unique_ptr<RNTupleInspector> Create(RNTuple *sourceNTuple);
+   static std::unique_ptr<RNTupleInspector> Create(std::string_view ntupleName, std::string_view storage);
 
-   /// Get the name of the RNTuple being inspected.
-   std::string GetName();
-
-   /// Get the number of entries in the RNTuple being inspected.
-   std::uint64_t GetNEntries();
+   /// Get the descriptor for the RNTuple being inspected.
+   RNTupleDescriptor *GetDescriptor() const { return fDescriptor.get(); }
 
    /// Get the compression settings of the RNTuple being inspected.
-   int GetCompressionSettings();
+   int GetCompressionSettings() const { return fCompressionSettings; }
 
    /// Get the on-disk, compressed size of the RNTuple being inspected, in bytes.
    /// Does **not** include the size of the header and footer.
-   std::uint64_t GetCompressedSize();
+   std::uint64_t GetOnDiskSize() const { return fOnDiskSize; }
 
-   /// Get the total, uncompressed size of the RNTuple being inspected, in bytes.
+   /// Get the total, in-memory size of the RNTuple being inspected, in bytes.
    /// Does **not** include the size of the header and footer.
-   std::uint64_t GetUncompressedSize();
+   std::uint64_t GetInMemorySize() const { return fInMemorySize; }
 
    /// Get the compression factor of the RNTuple being inspected.
-   float GetCompressionFactor();
+   float GetCompressionFactor() const { return (float)fInMemorySize / (float)fOnDiskSize; }
+
+   const RColumnInfo &GetColumnInfo(DescriptorId_t physicalColumnId) const;
+
+   const RFieldTreeInfo &GetFieldTreeInfo(DescriptorId_t fieldId) const;
+   const RFieldTreeInfo &GetFieldTreeInfo(std::string_view fieldName) const;
+
+   /// Get the number of fields of a given type or class present in the RNTuple.
+   /// TODO: Add regex support.
+   int GetFieldTypeCount(std::string_view typeName, bool includeSubFields = true) const;
```

Comment:
I think it's semantically more correct to return `size_t` instead (also in `GetColumnTypeCount()` below), 
```suggestion
   std::size_t GetFieldTypeCount(std::string_view typeName, bool includeSubFields = true) const;
```